File: app/api/models/blueprint/readme.rb
module App
  class Api
    module Models
      class Blueprint
        class Readme

          def initialize(blueprint)
            @blueprint = blueprint
          end

          def to_s
            return '' unless pathname.exist?
            pathname.read
          end

          def save(readme)
            pathname.write(readme)
          end

          def delete
            pathname.delete if exist?
          end

          def exist?
            pathname.exist?
          end

          private

          def pathname
            @blueprint.pathname.join('README.md')
          end
        end
      end
    end
  end
end

File: app/client/app/admin/blueprints/readme.js
app.admin.blueprints.readme = (route) => (a, x) => a['#app-blueprint-readme'](
  route.mount({
    routes: {
      '/?': route => app.admin.blueprints.readme.edit(route),
      '/delete': route => app.admin.blueprints.readme.delete(route),
    }
  })
)
